src/VisualStudio/Core/Def/LanguageService/AbstractPackage.cs

```diff
@@ -35,16 +35,17 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
         Assumes.Present(_componentModel_doNotAccessDirectly);
     }
 
-    protected override Task OnAfterPackageLoadedAsync(CancellationToken cancellationToken)
+    protected override async Task OnAfterPackageLoadedAsync(CancellationToken cancellationToken)
     {
         // TODO: remove, workaround for https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1985204
         var globalOptions = ComponentModel.GetService<IGlobalOptionService>();
         if (globalOptions.GetOption(SemanticSearchFeatureFlag.Enabled))
         {
+            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
Should this switch back? Otherwise it means the OnAfterPackageLoadedAsync is being called on varying threads, which _might_ be a surprise if the first code in there doesn't do some explicit switch itself.

Reply:
(or if that base method doesn't actually do anything, then maybe we just shouldn't call it in the first place)

Reply:
Reordered the calls.

Reply:
BTW, why do we use `ConfigureAwait(true)` here? https://github.com/dotnet/roslyn/pull/77195/files#diff-07f6b9e931d106ed073935f859e2101665ee01a84769742e2d6ea46d0578b7c2R30